### ISSUE TITLE
Support running V8 in JITless mode

### ISSUE THREAD

**Issue by Zoraver** [bromite/bromite#1235]
### Is your feature request related to privacy?
It is related to security, which is a prerequisite for privacy. Recall our conversation in https://github.com/GrapheneOS/Vanadium/pull/60.

### Is there a patch available for this feature somewhere?
Yes, support for *optionally* running with V8 JIT disabled was added in the latest version of Vanadium. See https://github.com/GrapheneOS/Vanadium/blob/11/patches/0079-JIT-less-toggle.patch for details.

### Describe the solution you would like

Ideally, I would like the patch in question from Vanadium to be adopted by Bromite.

### Describe alternatives you have considered

N/A

**Comment by uazo** [comment]
quoted from v8 blog:
`We encourage embedders on restricted platforms or with special security requirements to consider V8’s new JIT-less mode`

enabling that flag practically makes the memory pages non-executable (so decrease the attack surface) and disables the native optimization/compilation of js and webasm (also this, I've checked) at the cost of performance.
https://v8.dev/blog/jitless more details. and [here](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/heap/paged-spaces.h;l=222;drc=58bbaf417733a9ede1be0adb80a1ea927cb8b240;bpv=1;bpt=1) where disable execution flag

reading the code, the activation of the flag implies:

```
No turbofan compilation
Disable Field representation tracking (FLAG_track_heap_object_fields and FLAG_track_field_types)
Regexps are interpreted
No (experimental) Sparkplug compilation.
```
@csagan5 probably should be added default false as the actual patch.
maybe in the future we can think of an ad hoc content setting, as long as v8 is initialized after their management.

**Comment by csagan5** [comment]
> enabling that flag practically makes the memory pages non-executable (so decrease the attack surface) and disables the native optimization/compilation of js and webasm (also this, I've checked) at the cost of performance.

Yes, I was already going to include this on next release.

> as long as v8 is initialized after their management.

I did not understand this part.

**Comment by uazo** [comment]
>> as long as v8 is initialized after their management.

>I did not understand this part.

I mean that you have to check the execution order, if the part of code that transfers the information (via mojom) of the content settings to the rendering thread happens before the initialization of v8, we can add it easily.
i think so, because javascript may not be used by blink so it doesn't make sense to activate v8, but i don't know.

**Comment by csagan5** [comment]
Fixed in `91.0.4472.143`.

**Comment by wchen342** [comment]
@Eloston Will this be useful to be included in `ungoogled-chromium` too?
